src/parser/pegen.ts

```diff
@@ -1630,12 +1630,37 @@ function _set_name_context(e: Name, ctx: expr_context) {
 // }
 
 /* Creates an `expr_ty` equivalent to `expr` but with `ctx` as context */
-export function set_expr_context(expr: expr, ctx: expr_context): expr {
+type exprKind = typeof expr | typeof Name;
+
+export function set_expr_context(e: expr, ctx: expr_context): expr {
     assert(expr !== null);
     let newExpr: expr;
-    switch (expr._kind) {
+    switch (e.constructor as exprKind) {
+        case Name:
+            newExpr = _set_name_context(e as Name, ctx);
+            break;
```

Comment:
preferences? this vs the _kind. I think maybe this is better. And then we can revert the _kind idea.
we can generate `exprKind` in the asdl which will just be a union of the typeofs